ocaml/src/log_plugin.ml

```diff
@@ -28,41 +31,52 @@ let user_function (user_db : Registry.user_db) (value_o : string option) =
     let msgs = Llio.list_from Llio.string_from buf in
 
     let next_id = match user_db # get next_id_key with
-      | None -> 0l
-      | Some v -> deserialize Llio.int32_be_from v
+      | None -> int_0
+      | Some v -> deserialize int_be_from v
     in
     let new_next_id =
       List.fold_left
         (fun next_id msg ->
            user_db # put
-             (log_prefix ^ (serialize Llio.int32_be_to next_id))
+             (log_prefix ^ (serialize int_be_to next_id))
              (Some msg);
-           Int32.succ next_id)
+           int_succ next_id)
         next_id
         msgs in
     user_db # put
       next_id_key
-      (Some (serialize Llio.int32_be_to new_next_id));
+      (Some (serialize int_be_to new_next_id));
     None
 
-let name = "log_32"
+let user_function_32 user_db value_o =
+  user_function user_db value_o
+                (Llio.int32_be_from, Llio.int32_be_to, 0l, Int32.succ)
 
-let make_update ~next_id_key ~log_prefix ~msgs =
-  let payload =
-    serialize
-      (Llio.tuple3_to
-         Llio.string_to
-         Llio.string_to
-         (Llio.list_to Llio.string_to))
-      (next_id_key, log_prefix, msgs)
-  in
-  Update.Update.UserFunction (name, Some payload)
+let user_function_x64 user_db value_o =
+  user_function user_db value_o
+                (x_int64_be_from, x_int64_be_to, 0L, Int64.succ)
+
+let name_log_32 = "log_32"
+let name_log_x64 = "log_x64"
+
+let make_payload ~next_id_key ~log_prefix ~msgs =
+  serialize
+    (Llio.tuple3_to
+       Llio.string_to
+       Llio.string_to
+       (Llio.list_to Llio.string_to))
+    (next_id_key, log_prefix, msgs)
+
+let make_update_x64 ~next_id_key ~log_prefix ~msgs =
+  let payload = make_payload ~next_id_key ~log_prefix ~msgs in
+  Update.Update.UserFunction (name_log_x64, Some payload)
```

Comment:
This change of user function name forces that all nodes of the alba manager should be updated at the same time ...
So probably it's a better idea to keep the same name, which in newer version may write data in a different format (once id > Int32.max_int).

Hmm, now that I think of it: we should probably bump the albamgr version now too https://github.com/openvstorage/alba/blob/master/ocaml/src/albamgr_plugin.ml#L695 (as an old node may otherwise deserialize in an incorrect way leading to weird problems)?
(Again it would only happen when we actually ran out of work item ids)

So I guess that means they would want to update all abm nodes to the same version within some short timespan too.